### DISCUSSION TITLE
onContextMenu

### DISCUSSION THREAD

**Discussion by Maxim-BC** [dowjones/react-dropdown-tree-select#569] [Q&A]
Hello! I wanted to find out if there is an option in this library to use "onContextMenu" so that I can bring up the context menu by right-clicking? Thank you in advance!

**Comment by mrchief** [comment]
Do you mean you want an event that is fired when the context menu is shown? This component doesn't have a context menu so I'm not sure what you're asking.

**Comment by Maxim-BC** [comment]
I have a custom menu, I want it to be called by right-clicking on one of the elements of the tree. And in this menu were the value of the element on which I right-clicked

**Comment by mrchief** [comment]
You can add a right click handler to any of the tree elements - there are several classes and other properties you can use to attach right click handlers. 
![image](https://user-images.githubusercontent.com/781818/175117694-1ed322fc-dc81-4807-8d6d-0830be1ec79e.png)

Although I think we may need a code change to listen for `onContextMenu` in addition to `onClick` event and surface the click type. WIll you be interested in sending a PR?
